moz_security/sandboxes/heka/analysis/moz_security_hh.lua

```diff
@@ -0,0 +1,158 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+#  Mozilla Security Heavy Hitters
+
+See: https://cacm.acm.org/magazines/2009/10/42481-finding-the-frequent-items-in-streams-of-data/abstract
+
+For events matching the message_matcher, this analysis plugin identifies the number of events
+seen for a given identification field (e.g., the IP address in an nginx log). Can be utilized to trend
+heavy usage from specific identifiers. Event counts per identifier are stored in a count-min sketch.
+
+During each ticker interval, the plugin calculates the average requests per identifier within the
+window (event frequency); identifiers that have made requests that exceeed the average + the calculated
+threshold cap will be captured in the plugin output. Note the threshold is calculated on each interval
+using the formula shown in the sample configuration.
+
+The plugin by default outputs collected data as TSV; if the send_tigerblood configuration option
+is enabled the plugin will instead output data as Tigerblood violation messages (for consumption by
+for example a Tigerblood output module).
+
+Heavy hitters will be identified on each interval tick, so ensure the ticker_interval parameter is
+set to a value appropriate for consumption of the intended event stream to ensure the gathered sample
+is sufficient.
+
+## Sample Configuration
+```lua
+filename = "moz_security_heavy_hitters.lua"
+message_matcher = "Logger == 'input.nginx'"
+ticker_interval = 60
+preserve_data = false
+
+id_field = "Fields[remote_addr]" -- field to use as the identifier
+-- id_field_capture = ",? *([^,]+)$",  -- optional e.g. extract the last entry in a comma delimited list
+
+-- list_max_size = 10000 -- optional, defaults to 10000 (maximum number of heavy hitter IDs to track)
```

Comment:
The list was only this big for the frequent items calculation (you probably don't want more than 50 or 100 here).